src/EventListener/CategoriesListener.php

```diff
@@ -59,7 +59,7 @@ public function adaptChoicesField(DataContainer $dc = null): void
                 .'<a class="tl_submit" style="margin-top: 10px;" href="'.Backend::addToUrl('key=scale').'" title="'.StringUtil::specialchars($GLOBALS['TL_LANG']['tl_survey_question']['addscale'][1]).'" onclick="Backend.getScrollOffset();">'.StringUtil::specialchars($GLOBALS['TL_LANG']['tl_survey_question']['addscale'][0]).'</a><p style="height: 0;margin: 0;">';
         }
 
-        if (($surveyModel = SurveyModel::findByQuestionId($questionModel->id)) && $surveyModel->useResultCategories) {
+        if (($surveyModel = SurveyModel::findByQuestionId((int) $questionModel->id)) && $surveyModel->useResultCategories) {
```

Comment:
_Note:_ this error only occurs in PHP 7, not PHP 8, as the `id` will be an `integer` in PHP 8.